PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/LinkAzureServiceBeforeRunProvider.java

```diff
@@ -0,0 +1,192 @@
+import com.microsoft.azure.toolkit.intellij.link.mysql.MySQLConnectionUtils;
+import com.microsoft.azure.toolkit.intellij.link.mysql.PasswordConfig;
+import com.microsoft.azure.toolkit.intellij.link.mysql.PasswordDialog;
+import com.microsoft.azure.toolkit.intellij.link.po.BaseServicePO;
+import com.microsoft.azure.toolkit.intellij.link.po.LinkPO;
+import com.microsoft.azure.toolkit.intellij.link.po.MySQLServicePO;
+import com.microsoft.azure.toolkit.intellij.webapp.runner.webappconfig.IntelliJWebAppSettingModel;
+import com.microsoft.azure.toolkit.intellij.webapp.runner.webappconfig.WebAppConfiguration;
+import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel;
+import com.microsoft.intellij.AzureLinkStorage;
+import com.microsoft.intellij.AzureMySQLStorage;
+import com.microsoft.intellij.actions.SelectSubscriptionsAction;
+import com.microsoft.intellij.helpers.AzureIconLoader;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureIconSymbol;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+public class LinkAzureServiceBeforeRunProvider extends BeforeRunTaskProvider<LinkAzureServiceBeforeRunTask> {
+    private static final Logger LOGGER = Logger.getInstance(SelectSubscriptionsAction.class);
+    private static final String NAME = "Link Azure Service";
+    private static final String DESCRIPTION = "Link Azure Service Task";
+    private static final String SPRING_BOOT_CONFIGURATION_REF = "com.intellij.spring.boot.run.SpringBootApplicationRunConfiguration";
+    public static final Key<LinkAzureServiceBeforeRunTask> ID = Key.create("LinkAzureServiceBeforeRunProviderId");
+    public static final Key<Boolean> LINK_AZURE_SERVICE = Key.create("LinkAzureService");
+    public static final Key<Map<String, String>> LINK_AZURE_SERVICE_ENVS = Key.create("LinkAzureServiceEnvs");
+
+    @Override
+    public Key<LinkAzureServiceBeforeRunTask> getId() {
+        return ID;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getDescription(LinkAzureServiceBeforeRunTask task) {
+        return DESCRIPTION;
+    }
+
+    @Nullable
+    @Override
+    public Icon getIcon() {
+        return AzureIconLoader.loadIcon(AzureIconSymbol.Common.AZURE);
+    }
+
+    @Nullable
+    @Override
+    public LinkAzureServiceBeforeRunTask createTask(@NotNull RunConfiguration runConfiguration) {
+        boolean alwaysEnable = false;
+        if (StringUtils.equals(runConfiguration.getClass().getName(), SPRING_BOOT_CONFIGURATION_REF)
+        /*|| runConfiguration instanceof com.intellij.javaee.appServers.run.configuration.CommonStrategy*/) {
```

Comment:
why we don't use `instance of SpringBootApplicationRunConfiguration` here? For class loader issue?

Reply:
SpringBootApplicationRunConfiguration is only for IU.